stores/pg/src/storage/domains/observability/v-next/index.ts

```
@@ -289,21 +289,49 @@ export class ObservabilityStoragePostgresVNext extends ObservabilityStorage {
+      } catch (error) {
+        pruneFailed = true;
+        throw error;
+      } finally {
+        // Pruning is not atomic across signal tables. If a later table fails
+        // after an earlier one deleted rows, the monotonic tag cache still
+        // has to be rebuilt on its next read.
+        if (invalidatesTagDiscovery) {
+          try {
+            await discoveryOps.invalidateTagDiscoveryCache(this.#client, this.#schema);
+          } catch (error) {
+            if (!pruneFailed) throw error;
```

Biome (2.5.6)

[error] Unsafe usage of 'throw'.

(lint/correctness/noUnsafeFinally)